packages/web3-types/src/web3_base_provider.ts

```diff
@@ -157,28 +160,32 @@ export abstract class Web3BaseProvider<API extends Web3APISpec = EthExecutionAPI
 
 	public abstract on(
 		type: 'disconnect',
-		callback: Web3ProviderEventCallback<ProviderRpcError>,
+		callback: Web3Eip1193ProviderEventCallback<ProviderRpcError>,
+	): void;
+	public abstract on(
+		type: 'message' | string,
+		callback: Web3ProviderEventCallback<ProviderMessage>,
 	): void;
 	public abstract on<T = JsonRpcResult>(
 		type: 'message' | string,
 		callback: Web3ProviderEventCallback<T>,
 	): void;
 	public abstract on(
 		type: 'connect' | 'chainChanged',
-		callback: Web3ProviderEventCallback<ProviderConnectInfo>,
+		callback: Web3Eip1193ProviderEventCallback<ProviderConnectInfo>,
 	): void;
 	public abstract on(
 		type: 'accountsChanged',
-		callback: Web3ProviderEventCallback<{
+		callback: Web3Eip1193ProviderEventCallback<{
```

Comment:
seems that types are still wrong, if you check the [appendix of specification](https://eips.ethereum.org/EIPS/eip-1193#appendix-i-consumer-facing-api-documentation) you can observe different type

Reply:
maybe the wallet connect example can help visualize the issue
https://docs.walletconnect.com/1.0/quick-start/dapps/web3-provider#events-eip-1193